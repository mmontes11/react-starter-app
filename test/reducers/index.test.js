import deepFreeze from "deep-freeze";
import rootReducer from "reducers";

jest.mock("reducers/history", () => (state = "INITIAL_HISTORY", action) => {
  switch (action.type) {
    case "ACTION":
      return "STATE_HISTORY";
    default:
      return state;
  }
});

jest.mock("reducers/next", () => (state = "INITIAL_NEXT", action) => {
  switch (action.type) {
    case "ACTION":
      return "STATE_NEXT";
    default:
      return state;
  }
});

jest.mock("reducers/step", () => (state = "INITIAL_STEP", action) => {
  switch (action.type) {
    case "ACTION":
      return "STATE_STEP";
    default:
      return state;
  }
});

describe("reducers/index", () => {
  it("combines other reducers", () => {
    expect(rootReducer).toMatchSnapshot();
  });

  const state = {
    history: "INITIAL_HISTORY",
    xIsNext: "INITIAL_NEXT",
    step: "INITIAL_STEP",
  };

  deepFreeze(state);

  it("has a default state", () => {
    expect(rootReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(rootReducer(state, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces ACTION", () => {
    expect(rootReducer(state, { type: "ACTION" })).toMatchSnapshot();
  });
});
