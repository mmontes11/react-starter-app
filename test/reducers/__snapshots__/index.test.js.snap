// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/index combines other reducers 1`] = `[Function]`;

exports[`reducers/index has a default state 1`] = `
Object {
  "history": "INITIAL_HISTORY",
  "step": "INITIAL_STEP",
  "xIsNext": "INITIAL_NEXT",
}
`;

exports[`reducers/index reduces ACTION 1`] = `
Object {
  "history": "STATE_HISTORY",
  "step": "STATE_STEP",
  "xIsNext": "STATE_NEXT",
}
`;

exports[`reducers/index reduces nothing 1`] = `
Object {
  "history": "INITIAL_HISTORY",
  "step": "INITIAL_STEP",
  "xIsNext": "INITIAL_NEXT",
}
`;
